weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -4115,6 +4116,53 @@ def completions_create_stream(
         When req.inputs.n > 1, properly separates and tracks all choices
         within a single call's output rather than creating separate calls.
         """
+        # --- Resolve prompt if provided and prepend messages
+        prompt = getattr(req.inputs, "prompt", None)
+        template_vars = getattr(req.inputs, "template_vars", None)
+        prompt_messages = []
+        # Store original messages before template var replacement (for tracking)
+        initial_messages = getattr(req.inputs, "messages", None) or []
+
+        if prompt:
```

Comment:
seems like some of this logic (handling combining messages, etc...) might want to be shared with the LLMStructuredCompletionModel class